datalad/support/annexrepo.py

```diff
@@ -1683,16 +1688,19 @@ def check(files, **kwargs):
             return self.info([f for f in files if not isdir(f)],
                              fast=True, **kwargs)
 
+        # XXX near duplicate to helper in file_has_content()
         def quick_check(filename):
-            filepath = opj(self.path, filename)
+            filepath = self.pathobj / filename
             # todo checks for being not outside of this repository
             # Note: ben removed '.git/' from '.git/annex/objects',
             # since it is not true for submodules, whose '.git' is a
             # symlink and being resolved to some
             # '.git/modules/.../annex/objects'
             return (
-                islink(filepath)
-                and 'annex/objects' in str(realpath(filepath))  # realpath OK
+                filepath.is_symlink()
+                # XXX str(filepath) should be filepath.as_posix()
+                # but somehow it doesn't work
+                and 'annex/objects' in str(filepath.resolve())  # realpath OK
```

Comment:
Just a reminder that this is the same as line 1653. (So, if fixable there ...)

Reply:
Same comment also ;-)